src/Exercism.Analyzers.CSharp/Analyzers/WeighingMachine/WeighingMachineAnalyzer.cs

```diff
@@ -0,0 +1,66 @@
+﻿
+using Exercism.Analyzers.CSharp.Analyzers.Shared;
+
+using static Exercism.Analyzers.CSharp.Analyzers.Shared.SharedComments;
+using static Exercism.Analyzers.CSharp.Analyzers.WeighingMachine.WeighingMachineSolution;
+
+namespace Exercism.Analyzers.CSharp.Analyzers.WeighingMachine
+{
+    internal class WeighingMachineAnalyzer : SharedAnalyzer<WeighingMachineSolution>
+    {
+        protected override SolutionAnalysis DisapproveWhenInvalid(WeighingMachineSolution solution)
+        {
+            if (solution.MissingWeighingMachineClass)
+                solution.AddComment(MissingClass(WeighingMachineClassName));
+
+            foreach (var missing in solution.MissingRequiredProperties())
+            {
+                solution.AddComment(MissingProperty(missing));
+            }
+
+            if (!solution.PrecisionIsAutoProperty)
+            {
+                solution.AddComment(new SolutionComment("not PrecisionIsAutoProperty"));
```

Comment:
You'll probably want to create a public static member in the `WeighingMachineComments` class for this comment. Same with the comments below.